src/store/migrations/v1.js

```diff
@@ -0,0 +1,232 @@
+        key: internalKey,
+        data: encrypted.mnemonic,
+      }).trim()
+
+      const passphraseDec: string = encryption.decryptData({
+        key: internalKey,
+        data: encrypted.passphrase,
+      })
+
+      return {
+        ...item,
+        derivationIndex: 0,
+        xpub: bip32XPublicKey,
+        encrypted: {
+          ...encrypted,
+          xprv: encryption.encryptData({
+            key: internalKey,
+            data: getXPRVFromMnemonic(
+              mnemonicDec,
+              passphraseDec,
+              derivationPath,
+            ),
+          }),
+        },
+      }
+    }
+
+    default:
+      return null
+  }
+}
+
+export async function migratePasswordToV1(walletsStored: Object): Promise<PasswordPersistV1> {
+  const {
+    internalKey,
+    passwordOptions,
+  }: Object = walletsStored
+
+  if (
+    type.isVoid(internalKey) ||
+    !type.isObject(internalKey) ||
+    type.isVoid(passwordOptions) ||
+    !type.isObject(passwordOptions)
+  ) {
+    throw new TypeError('Inconsistent password data')
+  }
+
+  const {
+    salt,
+    passwordHint,
+  }: Object = passwordOptions
+
+  if (!type.isString(salt) || !type.isString(passwordHint)) {
+    throw new TypeError('Inconsistent password data')
+  }
+
+  return {
+    salt,
+    internalKey,
+    hint: passwordHint,
+  }
+}
+
+export async function migrateWalletsToV1(
+  password: string,
+  walletsStored: Object,
+  passwordPersist: PasswordPersistV1,
+): Promise<WalletsPersistV1> {
+  const {
+    items,
+    activeWalletId,
+  }: Object = walletsStored
+
+  const {
+    salt,
+    internalKey,
+  }: PasswordPersistV1 = passwordPersist
+
+  if (!internalKey) {
+    throw new Error('Inconsistent password data')
+  }
+
+  const derivedKey: Uint8Array = await encryption.deriveKeyFromPassword(
+    password,
+    salt,
+  )
+
+  const internalKeyDec: Uint8Array = encryption.decryptInternalKey(
+    internalKey,
+    derivedKey,
+  )
+
+  const itemsNew: WalletV1[] = items.map((item: any): ?WalletV1 => migrateWalletToV1(
+    item,
+    internalKeyDec,
+  )).filter((item: ?WalletV1): boolean => !!item)
+
+  return {
+    items: itemsNew,
+    activeWalletId: activeWalletId || itemsNew.length ? itemsNew[0].id : null,
+  }
+}
+
+export async function migrateToV1(password: string): Promise<void> {
+  const walletsStored: Object = await getStoreData(WALLETS_STORE_KEY)
+  const passwordPersist: PasswordPersistV1 = await migratePasswordToV1(walletsStored)
+
+  const walletsPersist: WalletsPersistV1 = await migrateWalletsToV1(
+    password,
+    walletsStored,
+    passwordPersist,
+  )
+
+  const transaction: IDBTransaction = await initTransaction()
+
+  await deleteStoreData(TRANSACTIONS_STORE_KEY, transaction)
+  await setStoreData(walletsPersist, WALLETS_STORE_KEY, transaction)
+  await setStoreData(passwordPersist, PASSWORD_STORE_KEY, transaction)
+  await setStoreVersion(STORAGE_VERSION, transaction)
```

Comment:
Maybe we should wrap all of this in try..catch statement with catch sending exception to GA? Or maybe we should do it in `index.js`, so it would work the same for future migrations?